channeldb/invoices.go

```diff
@@ -1050,6 +1059,11 @@ func putInvoice(invoices, invoiceIndex, payAddrIndex, addIndex kvdb.RwBucket,
 	i *Invoice, invoiceNum uint32, paymentHash lntypes.Hash) (
 	uint64, error) {
 
+	// Avoid conflicts with all-zeroes magic value in the database.
+	if paymentHash == unknownPreimage.Hash() {
```

Comment:
isn't it more appropriate to have this check in `validateInvoice`? and the one below?

Reply:
Moved this check into `validateInvoice`. The one below is also called on update invoice. Moving it to `validateInvoice` would require all invoice fields to be checked on update, while this serves no purpose.

Reply:
In theory we _should_ be doing that anyway given that the whole invoice is reserialized when updating. No guarantee that those fields weren’t modified to invalid values. But doing this is certainly out of scope for this PR

Reply:
In theory we _should_ be doing that anyway given that the whole invoice is reserialized when updating. No guarantee that those fields weren’t modified to invalid values. But doing this is certainly out of scope for this PR